### PR TITLE
Remove caserefgeneratorkey default

### PR DESCRIPTION
**Pull request by hbrace**
# Motivation and Context

We don't want to risk the caserefgeneratorkey being set to `abc123`.
# What has changed

Default value for the caserefgeneratorkey has been removed and now set in constructor so that startup fails if not present in configuration files. Tests updated.
# How to test?

Make build, run ATs
# Links

https://trello.com/c/PbLRX3Ln/391-caserefgeneratorkey-shouldnt-be-set-in-the-applicationyml-of-caseprocessor-3
# Screenshots (if appropriate):

### CODE REVIEW

**Review by ryangrundy7 — APPROVED** [review]
Looks good to me 👍

**Review by liamtoozer — COMMENTED** [review]
Should we also add the `caserefgeneratorkey` to our `application-dev.yml` for running locally with an IDE?

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java [hunk 1]
```diff
@@ -29,26 +29,27 @@
...
       EventLogger eventLogger,
-      RasRmCaseNotificationEnrichmentService rasRmNewBusinessCaseEnricher) {
+      RasRmCaseNotificationEnrichmentService rasRmNewBusinessCaseEnricher,
+      @Value("${caserefgeneratorkey}") byte[] caserefgeneratorkeybytes) {
     this.caseRepository = caseRepository;
     this.caseService = caseService;
     this.collectionExerciseRepository = collectionExerciseRepository;
     this.eventLogger = eventLogger;
     this.rasRmNewBusinessCaseEnricher = rasRmNewBusinessCaseEnricher;
+    this.caserefgeneratorkey = caserefgeneratorkeybytes.clone();
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java:52]: Why these changes?  It seemed to work fine previously and this make the test set up more awkward.

**Review by AdamHawtin — APPROVED** [review]
Works for me locally, fails to start in docker dev if I remove the config 👍
